source/connect/tls.txt

```diff
@@ -0,0 +1,241 @@
+
+OpenSSL
+~~~~~~~
+
+The MongoDB C Driver uses OpenSSL on Linux and Unix platforms (besides macOS). Industry best practices
+and some regulations require the use of TLS 1.1 or newer, which requires at least OpenSSL 1.0.1. Use the following command
+to check your OpenSSL version:
+
+.. code-block:: bash
+
+   $ openssl version
+
+Ensure your system's OpenSSL is a recent version (at least 1.0.1), or use the following command to install a recent version in a
+non-system path and build against it:
+
+.. code-block:: bash
+
+   cmake -DOPENSSL_ROOT_DIR=/absolute/path/to/openssl
+
+When compiled against OpenSSL, the driver will attempt to load the system default certificate store, as configured by the
+distribution. That can be overridden by setting the ``tlsCAFile`` URI option or with the fields ``ca_file`` and ``ca_dir``
+in the `mongoc_ssl_opt_t <{+api-libmongoc+}/mongoc_ssl_opt_t.html>`__.
+
+The Online Certificate Status Protocol (OCSP) (see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`__) is fully supported
+when using OpenSSL 1.0.1+ with the following notes:
+
+- When a ``crl_file`` is set with `mongoc_ssl_opt_t <{+api-libmongoc+}/mongoc_ssl_opt_t.html>`__ and the ``crl_file`` revokes
+  the server's certificate, the certificate is considered revoked (even if the certificate has a valid stapled OCSP response).
+
+LibreSSL / libtls
+~~~~~~~~~~~~~~~~~
+
+The MongoDB C Driver supports LibreSSL through the use of OpenSSL compatibility checks when configured to compile against
+``openssl``. It also supports the new ``libtls`` library when configured to build against ``libressl``.
+
+When compiled against the Windows native libraries, the ``crl_file`` option of a `mongoc_ssl_opt_t <{+api-libmongoc+}/mongoc_ssl_opt_t.html>`__ is not supported,
+and will issue an error if used.
+
+Setting ``tlsDisableOCSPEndpointCheck`` and ``tlsDisableCertificateRevocationCheck`` has no effect.
+
+The Online Certificate Status Protocol (OCSP) (see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`__) is partially supported
+with the following notes:
+
+- The Must-Staple extension (see `RFC 7633 <https://tools.ietf.org/html/rfc7633>`__) is ignored.
+- Connection will continue if a Must-Staple certificate is presented without a stapled response and the OCSP responder is down.
+- Connection will not continue if the client receives a revoked response from an OCSP responder.
+
+Native TLS Support on Windows (Secure Channel)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The MongoDB C Driver supports the Windows native TLS library (Secure Channel, or SChannel) and its native crypto library
+(Cryptography API: Next Generation, or CNG).
+
+When compiled against the Windows native libraries, the ``ca_dir`` option of a `mongoc_ssl_opt_t <{+api-libmongoc+}/mongoc_ssl_opt_t.html>`__ is not supported
+and will issue an error if used.
+
+Encrypted PEM files (e.g., setting ``tlsCertificateKeyPassword``) are also not supported and will result in error when
```

Comment:
S: can you explain what the options you mention in this section are? I'm guessing URI options
```suggestion
Encrypted PEM files (e.g., setting the ``tlsCertificateKeyPassword`` URI option) are also not supported and will result in error when
```